Knossos.NET/ViewModels/Templates/TaskItemViewModel.cs

```diff
@@ -204,6 +204,220 @@ public TaskItemViewModel()
+                                                    //Verify fail, clear list and break to next modversion
+                                                    copySrcList.Clear();
+                                                    copyDstList.Clear();
+                                                    break;
+                                                }
+                                            }
+
+                                            copySrcList.Add(oldPath);
+                                            var newPath = mod.devMode ? Path.Combine(mod.fullPath, package.folder!, f.filename) : Path.Combine(mod.fullPath, f.filename);
+                                            copyDstList.Add(newPath);
+                                        }
+                                        else
+                                        {
+                                            Log.Add(Log.LogSeverity.Information, "TaskItemViewModel.TryToCopyFilesFromOldVersions()", "Requested file "+oldPath+ " was not present on the old mod files. "+oldVer + " cant be used as source of files for "+file.filename);
+                                            //Verify fail, clear list and break to next modversion
+                                            copySrcList.Clear();
+                                            copyDstList.Clear();
+                                            break;
+                                        }
+                                    }
+                                    if (cancellationTokenSource.IsCancellationRequested)
+                                        throw new TaskCanceledException();
+                                }
+
+                                //we can know if the verify completed by checking the copySrcList
+                                if(copySrcList.Any() && copySrcList.Count() == copyDstList.Count())
+                                {
+                                    ProgressBarMax = copySrcList.Count();
+                                    //Copy files
+                                    for (int i = 0; i < copySrcList.Count(); i++)
+                                    {
+                                        //Make sure the dest folder structure exist
+                                        Directory.CreateDirectory(Path.GetDirectoryName(copyDstList[i])!);
+                                        //First lets try to hardlink, if one fails or it is disabled, revert to copy files
+                                        if (useHardlinks)
+                                        {
+                                            Info = "Hardlink file: " + (i + 1).ToString() + " / " + copySrcList.Count() + " (" + Path.GetFileName(copyDstList[i]) + ")";
+                                            useHardlinks = HardLink.CreateFileLink(copySrcList[i], copyDstList[i]);
+                                        }
+                                        if (!useHardlinks)
```

Comment:
So this means that when hardlinks fail once, it will not go back to trying hard links on the next file.  Is that the desired behavior?

Reply:
For the current package at least yes, that could only be caused by not having permissions or the filesystem not supporting hardlinks (FAT32, EXFAT).

Reply:
I would add a comment for that, then.

Reply:
Added in the function summary